packages/react-scripts/scripts/init.js

```diff
@@ -64,6 +101,10 @@ module.exports = function(appPath, appName, verbose, originalDirectory, template
     }
   });
 
+  if (gitInit()) {
```

Comment:
Good catch and a rather stupid mistake of me. Moved it to execute after adding react an and react-dom.